sktime/transformations/_delegate.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+"""Delegator mixin that delegates all methods to wrapped transformer.
+
+Useful for building estimators where all but one or a few methods are delegated.
+For that purpose, inherit from this estimator and then override only the methods
+    that are not delegated.
+"""
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["miraep8"]
+__all__ = ["_DelegatedTransformer"]
+
+from sktime.transformations.base import BaseTransformer
+
+
+class _DelegatedTransformer(BaseTransformer):
+    """Delegator mixin that delegates all methods to wrapped transformer.
+
+    Delegates inner transformer methods to a wrapped estimator.
+        Wrapped estimator is value of attribute with name self._delegate_name.
+        By default, this is "estimator_", i.e., delegates to self.estimator_
+        To override delegation, override _delegate_name attribute in child class.
+
+    Delegates the following inner underscore methods:
+        _fit, _transform, _inverse_transform, _update
+
+    Does NOT delegate get_params, set_params.
+        get_params, set_params will hence use one additional nesting level by default.
+
+    Does NOT delegate or copy tags, this should be done in a child class if required.
+    """
+
+    _delegate_name = "estimator_"
+
+    def _get_delegate(self):
+        return getattr(self, self._delegate_name)
+
+    def _fit(self, X, y=None):
+        """Fit transformer to X, optionally to y.
+
+        State change:
+            Changes state to "fitted".
+
+        Writes to self:
+            Sets is_fitted flag to True.
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : Series or Panel, any supported mtype
+            Data to fit transform to, of python type as follows:
+                Series: pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+                Panel: pd.DataFrame with 2-level MultiIndex, list of pd.DataFrame,
+                    nested pd.DataFrame, or pd.DataFrame in long/wide format
+                subject to sktime mtype format specifications, for further details see
+                    examples/AA_datatypes_and_datasets.ipynb
+        y : Series or Panel, default=None
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        self : a fitted instance of the estimator
+        """
+        estimator = self._get_delegate()
+        estimator.fit(X=X, y=y)
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_".
+            self._is_fitted
```

Comment:
also not true, only `transform` accesses `is_fitted`.

Reply:
Correct, also removed/modified.